OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string_analysis/interpretation/interprocedural/VirtualFunctionCallPreparationInterpreter.scala

```diff
@@ -0,0 +1,392 @@
+        var newValueSci = StringConstancyInformation.getNeutralElement
+        if (defSitesValueSci.isTheNeutralElement) {
+            val headSite = defSites.head
+            if (headSite < 0) {
+                newValueSci = StringConstancyInformation.lb
+            } else {
+                val ds = cfg.code.instructions(headSite).asAssignment.targetVar.usedBy.toArray.min
+                val r = exprHandler.processDefSite(ds, params)
+                // Again, defer the computation if there is no final result (yet)
+                if (r.isRefinable) {
+                    return r
+                } else {
+                    val p = r.asFinal.p.asInstanceOf[StringConstancyProperty]
+                    newValueSci = p.stringConstancyInformation
+                }
+            }
+        } else {
+            newValueSci = defSitesValueSci
+        }
+
+        val finalSci = param.value.computationalType match {
+            // For some types, we know the (dynamic) values
+            case ComputationalTypeInt ⇒
+                // The value was already computed above; however, we need to check whether the
+                // append takes an int value or a char (if it is a constant char, convert it)
+                if (call.descriptor.parameterType(0).isCharType &&
+                    defSitesValueSci.constancyLevel == StringConstancyLevel.CONSTANT) {
+                    if (defSitesValueSci.isTheNeutralElement) {
+                        StringConstancyProperty.lb.stringConstancyInformation
+                    } else {
+                        val charSciValues = sciValues.filter(_.possibleStrings != "") map { sci ⇒
+                            if (isIntegerValue(sci.possibleStrings)) {
+                                sci.copy(possibleStrings = sci.possibleStrings.toInt.toChar.toString)
+                            } else {
+                                sci
+                            }
+                        }
+                        StringConstancyInformation.reduceMultiple(charSciValues)
+                    }
+                } else {
+                    newValueSci
+                }
+            case ComputationalTypeFloat ⇒
+                InterpretationHandler.getConstancyInfoForDynamicFloat
+            // Otherwise, try to compute
+            case _ ⇒
+                newValueSci
+        }
+
+        val e: Integer = defSites.head
+        state.appendToFpe2Sci(e, newValueSci, reset = true)
+        FinalEP(e, StringConstancyProperty(finalSci))
+    }
+
+    /**
+     * Function for processing calls to [[StringBuilder#toString]] or [[StringBuffer#toString]].
+     * Note that this function assumes that the given `toString` is such a function call! Otherwise,
+     * the expected behavior cannot be guaranteed.
+     */
+    private def interpretToStringCall(
+        call: VirtualFunctionCall[V]
+    ): EOptionP[Entity, StringConstancyProperty] =
+        // TODO: Can it produce an intermediate result???
```

Comment:
add an assertion

Reply:
Not valid since dependency handing was revamped